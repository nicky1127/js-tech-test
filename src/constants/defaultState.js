const state = {
  liveEvents: [],
  event: {},
  markets: {},
  isOddsDecimal: false,
  loading: false,
  error: {},
  selections: []
};

export default state;
