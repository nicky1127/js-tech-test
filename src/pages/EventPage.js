import React, { useEffect, useState } from "react";
import clsx from "clsx";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core/styles";
import { connect } from "react-redux";
import MarketType from "components/accordions/MarketType";
import { getLiveEventList, getEventById } from "redux/_actions";
import { sortEventsByTime, groupEventsByType } from "redux/selectors";


const useStyles = makeStyles({
  root: {
    margin: "10px"
  },
  header: {
    color: "#fff",
    backgroundColor: "rgb(10,40,108)",
    borderTopLeftRadius: "5px",
    borderTopRightRadius: "5px"
  },
  icon: { color: "#fff" },
  details: {
    padding: 0,
    flexDirection: "column"
  }
});

const EventPage = props => {
  const classes = useStyles();

  const { event, markets } = props;

  let content;

  if (Array.isArray(markets) && markets.length > 0) {
    content = markets.map((market, index) => (
      <MarketType key={`marketType_${index}`} event={event} market={market} />
    ));
  }

  useEffect(() => {
    props.getLiveEventList(false);
    props.getEventById(21249945);
  }, []);

  return (
    <div className={clsx("eventPage", classes.root)}>
      {content}
    </div>
  );
};

EventPage.defaultProps = { event: {}, markets: {} };

EventPage.propTypes = {};

const mapStateToProps = (state, props) => {
  return {
    event: state.event,
    markets: Object.values(state.markets)
  };
};

const ConnectedEventPage = connect(mapStateToProps, {
  getLiveEventList,
  getEventById
})(EventPage);

export default ConnectedEventPage;
