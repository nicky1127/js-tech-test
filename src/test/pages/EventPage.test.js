import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import EventPage from "pages/EventPage";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import defaultState from "constants/defaultState";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = { location: { pathname: "test" } }; 

const state = {
  ...defaultState
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("EventPage", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <EventPage {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("EventPage should render one [data-testid=eventPage]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <EventPage {...props} />
      </Provider>
    );
    const container = getByTestId("eventPage");
    expect(container).toBeInTheDocument();
  });
});
