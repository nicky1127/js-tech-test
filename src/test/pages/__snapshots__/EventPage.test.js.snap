// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventPage Snapshot matchs 1`] = `
<div
  className="eventPage makeStyles-root-1"
  data-testid="eventPage"
>
  <div
    className="MuiBox-root MuiBox-root-10 makeStyles-detailContainer-4"
  >
    <span
      className="MuiTypography-root makeStyles-eventType-5 MuiTypography-body1"
    />
    <span
      className="MuiTypography-root makeStyles-eventDate-6 MuiTypography-body1"
    >
      Invalid Date Invalid Date
    </span>
    <div
      className="MuiGrid-root makeStyles-eventName-7 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-5"
        style={
          Object {
            "textAlign": "right",
          }
        }
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        V.S.
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-5"
        style={
          Object {
            "textAlign": "left",
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root makeStyles-eventScores-8 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-5"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        Home
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
        style={
          Object {
            "fontSize": "25px",
            "textAlign": "center",
          }
        }
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-5"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Away
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-11 makeStyles-marketContainer-9"
  />
</div>
`;
